main/getting-started/deploying.md

```diff
@@ -1,63 +1,94 @@
+- `contract/`which contains files relating to your smart contract itself.
+- `api/`which contains files enabling the UI frontend to communicate via HTTP/WebSocket to an on-chain backend contract instance and start your Dapp contract instance and backend.
+- `ui/` which contains files relating to your contract's user interface.
+
+## How it works
+
+All deployment happens via the local running Agoric process. This is usually the `ag-solo` process, 
+and frequently refered to as that or just as `ag-solo`. It is also sometimes described as/called an Agoric VM or a local server.
+
+`ag-solo` communicates with either a locally running or remote chain. The local process has a `home` object, which contains 
+references to services on-chain, including `zoe`, the default `registry`, and an application user's `wallet`. Developers can
+use these service references to call their associated API commands.
+
+Deploying to the chain first uploads the bundled contract source code to the local Agoric process (`ag-solo`).
+The deployment script then uses the `home` object to access `zoe` which installs the code on chain. 
+
+Via the REPL associated with the wallet, developers can use all the on-chain commands that deployment scripts use to deploy 
+contracts and Dapps.
+
+The `deploy.js` scripts run in an ephemeral `Node.js` outside of the Swingset kernel. 
+`ag-solo`'s spawner is persistent. Once the `deploy.js` script ends,
+connections to any of its objects are severed.
 
 ## Contract deployment
 
-The primary function of `contract/deploy.js` is to bundle up a contract's source code
-(which may consist of multiple files and modules) and "install" 
-it on the blockchain as source code, using `zoe`. This does _not_ execute 
-contract code; it just makes the code available on-chain.
+First, let's look at *contract deployment*. `contract/deploy.js`bundles up a contract's source code
+(which may consist of multiple files and modules) and "installs" 
+it on the blockchain as source code, using [`Zoe`](https://agoric.com/documentation/getting-started/intro-zoe.html). 
+This does _not_ execute contract code; it just makes the code available on-chain.
 
-The contract deployment process uses [`zoe.install`](https://agoric.com/documentation/zoe/api/zoe.html#e-zoe-install-code-moduleformat) to install 
-the source code on-chain. This returns an installation handle associated with the 
+The contract deployment process uses [`zoe.install()`](https://agoric.com/documentation/zoe/api/zoe.html#e-zoe-install-code-moduleformat) 
+to install the contract source code on-chain. This returns an *installation handle* associated with the 
 source code. In a typical contract deployment, the installation handle is stored 
-in the default shared registry so that it is broadly accessible on the chain.
-Its registry key is written to a config file in the `ui` directory. 
+in the default shared registry so it is broadly accessible on the chain.
```

Comment:
I believe that Zoe does not know anything about the registry. The deploy script gets and installation from Zoe, and then adds that to the registry (and then records the registry ID).